src/Form/MetadataDisplayForm.php

```diff
@@ -11,69 +19,205 @@
-    } else {
+    }
+    else {
       $this->messenger()->addMessage($this->t('The Metadata Display %entity has been added.', ['%entity' => $entity->toLink()->toString()]));
     }
-    \Drupal::service('plugin.manager.field.formatter')->clearCachedDefinitions();
-
-    /** @var \Drupal\Core\Template\TwigEnvironment $environment */
-    $environment = \Drupal::service('twig');
-    $environment->invalidate();
+    $this->formatterPluginManager->clearCachedDefinitions();
+    $this->twig->invalidate();
 
-    $form_state->setRedirectUrl($this->entity->toUrl('collection'));
     return $status;
   }
+
+  /**
+   * AJAX callback.
+   */
+  public static function ajaxPreview($form, FormStateInterface $form_state) {
+    $response = new AjaxResponse();
+
+    /** @var \Drupal\format_strawberryfield\MetadataDisplayInterface $entity */
+    $entity = $form_state->getFormObject()->getEntity();
+
+    // Attach the library necessary for using the OpenOffCanvasDialogCommand and
+    // set the attachments for this Ajax response.
+    $form['#attached']['library'][] = 'core/drupal.dialog.off_canvas';
+    $form['#attached']['library'][] = 'codemirror_editor/editor';
+    $response->setAttachments($form['#attached']);
+
+    $input = $form_state->getUserInput();
+    $id = EntityAutocomplete::extractEntityIdFromAutocompleteInput($input['ado_context_preview']);
+    /** @var \Drupal\node\NodeInterface $preview_node */
+    $preview_node = \Drupal::entityTypeManager()
+      ->getStorage('node')
+      ->load($id);
+    if (empty($preview_node)) {
+      return $response;
+    }
+
+    $sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($preview_node);
+
+    // Set initial context.
+    $context = [
+      'node' => $preview_node,
+      'iiif_server' => \Drupal::service('config.factory')->get('format_strawberryfield.iiif_settings')->get('pub_server_url'),
+    ];
+
+    // Add the SBF json context.
+    // @see MetadataExposeDisplayController::castViaTwig()
+    foreach ($sbf_fields as $field_name) {
+      /** @var \Drupal\strawberryfield\Field\StrawberryFieldItemList $field */
+      $field = $preview_node->get($field_name);
+      foreach ($field as $offset => $fielditem) {
+        $jsondata = json_decode($fielditem->value, TRUE);
+        // Preorder as:media by sequence.
+        $ordersubkey = 'sequence';
+        foreach (StrawberryfieldJsonHelper::AS_FILE_TYPE as $key) {
+          StrawberryfieldJsonHelper::orderSequence($jsondata, $key, $ordersubkey);
+        }
+        if ($offset === 0) {
+          $context['data'] = $jsondata;
+        }
+        else {
+          $context['data'][$offset] = $jsondata;
+        }
+      }
+    }
+    // Ensure we're using the twig from user's input instead of the entity's
+    // default.
+    $entity->set('twig', $input['twig'][0]);
+    $output = [
```

Comment:
Cool.  I feel this pull is almost ready, only one missing need right now and I'm totally open to suggestions. Ping @alliomeria too.
As explained in the ISSUE description we need to show the user the RAW JSON too (like as as in the twig fiddle example) because it is the `$context` they need to be able to pick values and do the actual Twig template writing. This fits (render array) here but we could also maybe use a horizontal/vertical tab to allow both to be displayed using the same height and selectable? Or some other UI/UX idea? Access to the RAW JSON as one writes/codes templates on the right side is quite important (in terms of interface). If having both there is too complex I feel we could even move the Preview itself down and give the RAW values priority. Thoughts you both?

Reply:
Hello @pcambra & @DiegoPino! Having the selected ADO's raw json available for quick review while writing Twig templates would be very helpful. A section (maybe collapsable?) above the new Preview area on the right hand side could be a comfortable setup.

Reply:
I am a bit confused about this, you're saying that we need somewhere to add the json from the SBF? I will try to implement what you're suggesting, let's see if I got it right

Reply:
Yes Pedro. It is in the ISSUE description and its super important for the editing experience. Without that you need to have another window with the RAW data visible because Twig templating will vary if a Key is there, if the value is single or plural. What I do not know is if 3 textures is the best OR some type of collapsible or even a replacement (a swap between JSON and the Preview). Mostly because once you see the Preview, you are Kinda Done with it until you press "preview button again". I will leave this decision to @alliomeria since she is the one that uses this most and can speak for the metadata community. Thanks!

Reply:
Your new code addresses this 🥇 . Thanks Pedro! Marking this as resolved!!